docs/DeveloperGuide.md

```diff
@@ -1,34 +1,890 @@
+      - [F.3. Search for ingredient](#f3-search-for-ingredient)
+      - [F.4. Search for recipe](#f4-search-for-recipe)
+      - [F.5. Search for chore](#f5-search-for-chore)
+      - [F.6. Saving data](#f6-saving-data)
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+## 1. Introduction
+### 1.1. Purpose
+The document contains the specified architecture and software design specifications for the application, Kitchen Helper. 
+### 1.2. Scope
+This describes the software architecture and software design requirements for Kitchen Helper. This guide is mainly for developers, designers and software engineers that are or going to work on Kitchen Helper. 
+## 2. Setting up
 
+### 2.1. Prerequisites
+1. JDK `11`.
+2. IntelliJ IDE.
 
-## Product Scope
-### Target user profile
+### 2.2. Setting up the project in your computer
+1. Fork this repository, and clone the fork repository to your computer.
+2. Open Intellij (if you are not in the welcome screen, click `File` > `Close Project` to close the existing project dialog first).
+3. Set up the correct JDK version for Gradle  
+   1. Click `Configure` > `Structure for New Projects` and then `Project Settings` > `Project` > `Project SDK`.
+   2. If `JDK 11` is listed in the drop down, select it. Otherwise, click `New…` and select the directory where you installed `JDK 11`.
+   3. Click `OK`.
+4. Click `Import Project`.
+5. Locate the `build.gradle` file and select it. Click `OK`.
+6. Click `Open as Project`.
+7. Click `OK` to accept the default settings if prompted. 
 
-{Describe the target user profile}
+## 3. Design
+This section provides a high level overview of our application, Kitchen Helper.
+### 3.1. Architecture
 
-### Value proposition
+![Architecture](images/KitchenHelperMain.png)
+The image above explains the design of the application, Kitchen Helper. 
 
-{Describe the value proposition: what problem does it solve?}
+The main driver of the application is `Main: Kitchen Helper`. It is responsible for mainly two phases:
+- At application launch
+    - This class will initialise the components in the correct sequence and is in charge of connecting them with each other.
+- At shut down
+    - This class will invoke cleanup method for the components when necessary.
+    
+In addition to that, the architecture of Kitchen Helper is broken down into seven classes, mainly the following: 
+- `Ui`: This class mainly handles the UI of the application.
+- `Parser`: This class mainly handles the parsing and handling of user commands.
+- `Command`: This class handles the type of command.
+- `Ingredient`: This class manages the data of data type ingredient in memory.
+- `Chore`: This class manages the data of data type chore in memory.
+- `Recipe`: This class manages the data of data type recipe in memory.
+- `Storage`: This class reads data from and writes data back into a text file for future uses.
 
-## User Stories
+### 3.2. Ui Component
```

Comment:
May I just check if Ui is supposed to be a class or interface? Based on the code below, Ui is a class but in the Ui component diagram it is labelled as an interface. Perhaps you can consider how to make some changes to your Ui component diagram to make it clearer to the user what Ui is suppose to be.

Reply:
The `Ui` class is in a `Ui` interface hence the `<<interface>>UI` pointing to `Ui`